docs/product/specs/permissions-revamp.md

```diff
@@ -0,0 +1,146 @@
+* Admins should be able to create new roles within a database page. 
+  - This will require the role used by the admin to have CREATEROLE privilege.
+  - The new  roles will be created on the database server that the database belongs to and have CONNECT and CREATE privileges on the database.
+  - The new roles will also have SELECT access to all items on the database that the role used by the admin has.
+* Future considerations:
+  - Roles exist on the database server and not under a database.
+  - However, inorder to simplify the user flow, we are not introducing a UX dedicated for database servers.
+  - We can explore this option in the future, based on user feedback or product strategy discussions.
+
+### Access control options
+#### Configuration
+* Admins should be able to provide Mathesar users access to each database configured in Mathesar.
+* Each standard Mathesar user should only be associated with a single role per database.
+  - Admins should be able to update this user-to-role configuration.
+* All Mathesar admins should be able to access all databases configured in Mathesar.
+  - See 'Role for admins' section on how roles are used by Admins.
+* Admins cannot be removed from a database.
+* Standard Mathesar users can be added/removed from a database.
+* Standard Mathesar users will not be able to access a database without being provided explicit access by a Mathesar admin.
+
+#### Permission checks for objects
+* The permissions for DB objects will be determined by the priviliges of the role the user is associated with.
+* For Mathesar specific objects, we will have a setting per user per database that allows the user to 'read' or 'read-write'.
+  - This applies for explorations and metadata such as table and column settings.
+  - Future consideration:
+    - We can allow an additional option 'Infer from DB privileges' that will automatically identify access to each Mathesar-object based on the privileges on the underlying DB objects.
+
+#### Role for admins
+* An admin is not associated with a single role.
+* Admins should be able to utilize any role that's configured for a database.
+* When an admin opens a DB object, lets say a table,
+  - If the owning role of that table is configured, then any operation performed by the admin will utilize that role.
+  - If the owing role isn't configured, we'll try to figure out the best role for the admin to use. In cases where it's not possible, we will use the `default` role.
+* More information is provided in the 'Default role' and 'Dealing with Ownership and Collaboration issues' sections of this document.
+
+### Default role
+* There should be one default role per database.
+* When a database is first configured, the role used to connect to the database will be set as the default role.
+  - This role is used for installing/upgrading the functions on the DB.
+* The default role cannot be removed.
```

Comment:
This implies you mean the credential instance. If correct, we should make that obvious in the document.